packages/contracts-bedrock/justfile

```diff
@@ -45,6 +45,22 @@ clean:
 test *ARGS: build-go-ffi
   forge test {{ARGS}}
 
+# Runs upgrade path variant of contract tests.
+# TODO: enforce that the op-deployer binary exists
+test-upgrade *ARGS: build-go-ffi
+  #!/bin/bash
+  if [ ! -f "../../op-deployer/bin/op-deployer" ]; then
+    echo "Error: op-deployer binary not found at ../../op-deployer/bin/op-deployer, you need to build it first"
+    exit 1
+  fi
+  cp ../../op-deployer/bin/op-deployer scripts/op-deployer
+  echo "Initializing intent.toml and state.json files in test/fixtures"
+  ./scripts/op-deployer init --l1-chain-id 1 --l2-chain-ids 271828 --workdir test/fixtures
+  UPGRADE_TEST=true \
+    DEPLOY_CONFIG_PATH=deploy-config/hardhat.json \
+    DEPLOYMENT_OUTFILE=deployments/upgrade-test-fork.json \
+    forge test {{ARGS}}
```

Comment:
Two comments here:
1. Why do we still need deploy config files? Are we deprecating them?
2. Nit, would be nice to have defaults for `DEPLOY_CONFIG_PATH` and `DEPLOYMENT_OUTFILE` that way the minimal diff between `just test` and `just test-upgrade` is the `UPGRADE_TEST=true` flag

Reply:
1. We should deprecate them yes, but for now the main test path is using hardhat.json to setup the system, so this additional test path should do the same. I think it would be a nice to have to include transitioning away from them as part of this work. 
2. Ack, I can likely hardcode those values in the solidity rather than using env vars.